cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -1023,42 +1034,44 @@ runQueryStakePools
   -> NetworkId
   -> Maybe OutputFile
   -> ExceptT ShelleyQueryCmdError IO ()
-runQueryStakePools (AnyConsensusModeParams cModeParams)
-                          network mOutFile = do
-  SocketPath sockPath <- firstExceptT ShelleyQueryCmdEnvVarSocketErr
-                           $ newExceptT readEnvSocketPath
+runQueryStakePools (AnyConsensusModeParams cModeParams) network mOutFile = OO.runOops1 @ShelleyQueryCmdError $ do
```

Comment:
How `runQueryStakePools` is implemented using `oops`.

Reply:
For comparison https://github.com/input-output-hk/cardano-node/pull/4788/files#r1071976898

Reply:
With `oops`:

```
runQueryStakePools
  :: AnyConsensusModeParams
  -> NetworkId
  -> Maybe OutputFile
  -> ExceptT ShelleyQueryCmdError IO ()
runQueryStakePools (AnyConsensusModeParams cModeParams) network mOutFile = OO.runOops1 @ShelleyQueryCmdError $ do
  SocketPath sockPath <- liftIO readEnvSocketPath
    & OO.leftM @EnvSocketError (\e -> OO.throwM $ ShelleyQueryCmdEnvVarSocketErr e)

  let localNodeConnInfo = LocalNodeConnectInfo cModeParams network sockPath

  result <- executeLocalStateQueryExpr_ localNodeConnInfo Nothing
    ( do
        anyE@(AnyCardanoEra era) <- case consensusModeOnly cModeParams of
          ByronMode -> return $ AnyCardanoEra ByronEra
          ShelleyMode -> return $ AnyCardanoEra ShelleyEra
          CardanoMode -> queryExpr_ $ QueryCurrentEra CardanoModeIsMultiEra

        let cMode = consensusModeOnly cModeParams

        eInMode <- toEraInMode era cMode
          & OO.throwNothingAsM (ShelleyQueryCmdEraConsensusModeMismatch (AnyConsensusMode cMode) anyE)

        sbe <- getSbeInQuery_ $ cardanoEraStyle era

        (queryExpr_ $ QueryInEra eInMode $ QueryInShelleyBasedEra sbe $ QueryStakePools)
          & OO.leftM (\e -> OO.throwM (ShelleyQueryCmdEraMismatch e)))

    & OO.catchM @AcquireFailure (\e -> OO.throwM $ ShelleyQueryCmdAcquireFailure $ toAcquiringFailure e)
    & OO.catchM @UnsupportedNtcVersionError (\e -> OO.throwM $ ShelleyQueryCmdUnsupportedNtcVersion e)

  writeStakePools_ mOutFile result
```

Reply:
I've simplified the `oops` library:

* The combinator names no longer end in `M`
* Adopted familiar function names like `hoistMaybe` that are found in other libraries
* Taken in account discussion from other PRs regarding naming

For more information, see https://github.com/haskell-works/oops/pull/11